tests/basic/storage_pool_test.go

```diff
@@ -9894,3 +9898,219 @@ var _ = Describe("{AddDriveWithKernelPanic}", func() {
+		}
+	}
+	return numofDrivesInNode
+}
+
+var _ = Describe("{AddDriveBeyondMaxSupported}", func() {
+
+	/*
+		Add Drive using legacy add drive feature
+		test max drive per pool
+		test max pool per node
+		test max drives per node
+	*/
+	var testrailID = 19170501
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/19170501
+	var runID int
+	JustBeforeEach(func() {
+		StartTorpedoTest("AddDriveBeyondMaxSupported", "Initiate pool Cloud add-drive beyond permitted and expect error", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+
+	stepLog := "Test max limit of pool,node and max pool in a cluster"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("pooladddrivebeyondmax-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+		stNodes := node.GetStorageNodes()
+		dash.VerifyFatal(len(stNodes) > 0, true, "Storage nodes found?")
+		selectedNode := stNodes[0]
+		if selectedNode.Id == "" {
+			log.FailOnError(fmt.Errorf("failed to get node id"), "failed to get node id")
+		}
+		maxDriveLimit, err := GetPoolMaxCloudDriveLimit(&selectedNode)
+		log.FailOnError(err, "failed to get pool drive limit")
+		stepStr := "Test beyond max drives per pool"
+		Step(stepStr, func() {
+			log.InfoD(stepStr)
+			var selectedPool *api.StoragePool
+			var maxDrivesAllowed int
+			//select a pool to add drives
+			poolListForOps, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			for i := 0; i < len(poolListForOps); i++ {
+				drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+				log.FailOnError(err, "error in getting pool with IO with error")
+				if drvs, ok := drvMap[fmt.Sprintf("%d", poolListForOps[i].ID)]; ok {
+					maxDrivesAllowed = POOL_MAX_CLOUD_DRIVES - len(drvs)
+					numofDrivesInNode := GetNumOfDrivesInNode(selectedNode)
+					//check if drives added  to pool will more than permissible for the node
+					if (int(maxDriveLimit) - numofDrivesInNode) > maxDrivesAllowed {
+						selectedPool = poolListForOps[i]
+						break
+					}
+				}
+			}
+			isjournal, err := IsJournalEnabled()
+			log.FailOnError(err, "Failed to check is journal enabled")
+			for i := 1; i <= maxDrivesAllowed+1; i++ {
+				drvSize, err := getPoolDiskSize(selectedPool)
+				log.FailOnError(err, "failed while getting pool size")
+				driveSize := drvSize * uint64(i)
+				expectedSize := (selectedPool.TotalSize / units.GiB) + driveSize
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+				log.FailOnError(err, "error while expanding pool")
+				resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+				if i == maxDrivesAllowed+1 {
```

Comment:
I believe control will never go inside this loop  , please check

Reply:
Its going i checked